### PR TITLE
Bug fixes and common API unifications

### PR DESCRIPTION
**Pull request by AlyShmahell**
1- converted randomness initialization to unseeded np.random.RandomState(), this will force the random number generator to take from /dev/urandom on linux (or from its equivalent on other systems) and achieve strong randomness.
2- converted the random number generator to a stream object, as a best practice for future expansion where multiple randomness objects with different random states are needed.
3- moved the teleportation example to its own directory to declutter the workspace.
4- converted the teleportation example to a bottlepy base, as the flask example wasn't running successfully, then added a usage doc-string for the user.
5- unified the examples' imports to "import kite as kt", then used "kt.<some_function/class>" as a common method of usage.
6- unified the library's import from explicit relative imports to Absolute imports for better readability, e.g. "from kite.<module> import <class>"

### CODE REVIEW

File: kite/tests.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import numpy as np
+from kite.gates import Gates
+
+
+class test_gates(unittest.TestCase):
+    
+    def test_base_gate_matrices(self):
+        "Test the gates which are hardcoded matrices"
+        g = Gates()
+        assert np.isclose(g.I, np.eye(2)).all()
```
Comment (vtomole) [inline kite/tests.py]: I'm not sure if  we need to test hardcoded matrices. If this test was already there, we need to delete it.

File: kite/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-from kite.program import *
+from kite.program import *
```
Comment (vtomole) [inline kite/__init__.py]: Need a newline

File: examples/teleportation/bob_teleport.qasm [hunk 3]
```diff
@@ -6,4 +6,4 @@ CNOT 1 2
 CLASSIC 1 1 1
 X 2
 CLASSIC 0 1 1
-Z 2 
+Z 2
```
Comment (vtomole) [inline examples/teleportation/bob_teleport.qasm]: Need a newline.

**Review by vtomole — CHANGES_REQUESTED** [review]
Thanks a lot for the Pull Request! Only 3 minor changes requested.

File: kite/tests.py (hunk 1 above)
Comment (AlyShmahell) [reply kite/tests.py]: I see it now, I put "pass" instead, but left the test module there for the future as a best practice.

File: kite/__init__.py (hunk 2 above)
Comment (AlyShmahell) [reply kite/__init__.py]: yep, always forgetting that, newline is added now.

**Review by vtomole — APPROVED** [review]

File: examples/teleportation/bob_teleport.qasm (hunk 3 above)
Comment (AlyShmahell) [reply examples/teleportation/bob_teleport.qasm]: newline is added here as well.

**Pull request merged**